### ISSUE TITLE
Python pip builder resolve dependency error : numpy==1.20.3

### ISSUE THREAD

**Issue by vynu** [aws/aws-lambda-builders#233]
**Description:**
Python pip builder resolve dependency error : numpy==1.20.3 while running sam build with pandas in lambda app requirements.txt

deep debugging :

ran `pip download --only-binary=:all: --no-deps --platform manylinux2014_x86_64 --implementation cp --abi cp38 --dest /tmp/samcli/scratch numpy==1.20.3`  by logging into actively running ` amazon/aws-sam-cli-build-image-python3.8`

its downloading with wheel file name : `Saved /tmp/samcli/scratch/numpy-1.20.3-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl`

I guess its failing at : [this-line](https://github.com/aws/aws-lambda-builders/blob/master/aws_lambda_builders/workflows/python_pip/packager.py#L346)

and I've tried with `pip download --only-binary=:all: --no-deps --platform manylinux2014_x86_64 --implementation cp --abi cp38 --dest /tmp/samcli/scratch numpy==1.20.1`

this downloaded wheel file : `Saved /tmp/samcli/scratch/numpy-1.20.1-cp38-cp38-manylinux2010_x86_64.whl`

clearly numpy==1.20.1 does not have any issue in file name

**Steps to reproduce the issue:**
1. Run `sam build  --use-container --build-image amazon/aws-sam-cli-build-image-python3.8 --debug` with pandas lib dependency in lambda app requirements.txt

**Observed result:**
`PythonPipBuilder:ResolveDependencies - {numpy==1.20.3(wheel)}`

**Expected result:**
`build successful`

**Additional environment details (Ex: Windows, Mac, Amazon Linux etc)**
`linux container env : amazon/aws-sam-cli-build-image-python3.8`

**Comment by vynu** [comment]
FYI:

adding `numpy==1.20.1` to your requirements.txt before `pandas` will not raise any problem

**Comment by charris** [comment]
See https://github.com/numpy/numpy/issues/18988, the problem is that the naming scheme changed in the manylinux docker images. The immediate AWS problem is:
```
    _MANYLINUX_COMPATIBLE_PLATFORM = {
        "any",
        "linux_x86_64",
        "manylinux1_x86_64",
        "manylinux2010_x86_64",
        "manylinux2014_x86_64",
    }
```
The set needs to include the new aliased names.

**Comment by aahung** [comment]
pinning the issue to prevent duplicates.

**Comment by aahung** [comment]
1.4.0 supports PEP 600 tags.

**Comment by kannanprasanna** [comment]
@aahung , any update on when next aws-sam-cli version will be released with aws-lambda-builders pointing to 1.4.0 (currently seems to be hardcoded to version 1.3.0) or do you recommend pinning aws-lambda-builders 1.4.0 in our requirements file?

**Comment by aahung** [comment]
[AWS SAM CLI 1.24.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.24.0) has been released now
